solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,668 @@
+    error_message = "ttl must match regular expression /^[0-9]+[s,m,h,d]{0,1}$/"
+  }
+}
+
+variable "exclude_cn_from_sans" {
+  type        = bool
+  description = "Whether the common name is excluded from Subject Alternative Names (SANs). If set to true, the common name is not included in DNS or Email SANs if they apply"
+  default     = false
+}
+
+##############################################################################
+# ROOT CA
+##############################################################################
+
+variable "root_ca_name" {
+  type        = string
+  description = "The name of the Root CA to be created for a private_cert secret engine."
+  default     = "root-ca"
+
+  validation {
+    condition     = length(var.root_ca_name) >= 2
+    error_message = "length of root_ca_name must be >= 2"
+  }
+}
+
+variable "root_ca_max_ttl" {
+  type        = string
+  description = "Maximum TTL value for the root CA"
+  default     = "87600h"
+
+  validation {
+    condition     = can(regex("^[0-9]+[s,m,h,d]{0,1}$", var.root_ca_max_ttl))
+    error_message = "root_ca_max_ttl must match regular expression /^[0-9]+[s,m,h,d]{0,1}$/"
+  }
+}
+
+variable "root_ca_common_name" {
+  type        = string
+  description = "Fully qualified domain name or host domain name for the certificate which is to be created"
+  default     = "terraform-modules.ibm.com"
+
+  validation {
+    condition     = length(var.root_ca_common_name) >= 4 && length(var.root_ca_common_name) <= 128
+    error_message = "length of root_ca_common_name must be >= 4 and <= 128"
+  }
+
+  validation {
+    condition     = can(regex("(.*?)", var.root_ca_common_name))
+    error_message = "root_ca_common_name must match regular expression /(.*?)/"
+  }
+}
+
+variable "root_ca_crl_expiry" {
+  type        = string
+  description = "The expiry time for root CA Certificate Revocation List (CRL)"
+  default     = null
+  validation {
+    condition     = var.root_ca_crl_expiry == null ? true : can(regex("^[0-9]+[s,m,h,d]{0,1}$", var.root_ca_crl_expiry))
+    error_message = "root_ca_crl_expiry must match regular expression /^[0-9]+[s,m,h,d]{0,1}$/"
+  }
+}
+
+variable "root_ca_crl_disable" {
+  type        = bool
+  description = "crl_disable flag for the root CA"
+  default     = false
+}
+
+variable "root_ca_crl_distribution_points_encoded" {
+  type        = bool
+  description = "crl_distribution_points_encoded flag for the root CA"
+  default     = true
+}
+
+variable "root_ca_issuing_certificates_urls_encoded" {
+  type        = bool
+  description = "issuing_certificates_urls_encoded flag for the root CA"
+  default     = true
+}
+
+##############################################################################
+# Intermediate CA
+##############################################################################
+
+variable "intermediate_ca_name" {
+  type        = string
+  description = "Name of the Intermediate CA to create for a private_cert secret engine"
```

Comment:
the description needs to be updated to mention about the naming convention used when using prefix